android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -0,0 +1,119 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.net.Uri
+import androidx.activity.result.ActivityResultCaller
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.*
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.util.concurrent.TimeUnit
+
+fun LocalAssociationScenario.begin(): MobileWalletAdapterClient {
+    return start().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+fun LocalAssociationScenario.end() {
+    close().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+@Suppress("BlockingMethodInNonBlockingContext")
+class MobileWalletAdapter(
+    private val resultCaller: ActivityResultCaller,
+    private val scenarioType: ScenarioTypes = ScenarioTypes.Local,
+    private val timeout: Int = Scenario.DEFAULT_CLIENT_TIMEOUT_MS,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) {
+    private val contract = LocalAssociationIntentContract(LocalAssociationIntentCreator())
+    private val resultLauncher = resultCaller.registerForActivityResult(contract) { }
+
+    private val scenarioChooser = ScenarioChooser()
+
+    suspend fun authorize(identityUri: Uri, iconUri: Uri, identityName: String): Boolean {
+        return withScenario(associate()) { client ->
```

Comment:
Definitely, I can see the RxJava APIs using some of these pieces as well, and thus it could make sense to port at least some of these changes to the "core" clientlib.